### PR TITLE
[pthreadpool] create a new port

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?

There was no port request for this project.

This is one of the 3rd party libraries for [the PyTorch project](https://github.com/pytorch/pytorch/blob/master/.gitmodules). The PR will be used for future support of the `libtorch` port.

### Which triplets are supported/not supported? Have you updated the CI baseline?

The project has quite simple code. Expected to be portable according to its CI configuration, but need more tests.

* `x64-osx`
* `x64-linux`
* `x64-windows`, `x86-windows`, `arm64-windows`: runs static build on Windows

#### Work In Progress

* Android/iOS triplets

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

The library has a tag, but it's too old. I snapshoted the source code with `"version-string": 2021-02-21`
 
* PR Structure
    * [x] Make separate Pull Requests per port.  **The PR requires #16341**
* Manifests and CONTROL files
    * [x] `vcpkg format-manifest`
* Versioning
    * [x] Follow common conventions for the "version" field  
      [... we use the date that the commit was accessed by you, formatted as `YYYY-MM-DD`.](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md#version-fields)

### CODE REVIEW

**Comment by luncliff** [comment]
This PR will be rebased after #16341 is merged. Will be ready for review after testing more triplets.

File: ports/pthreadpool/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "pthreadpool",
+  "version-string": "2020-02-21",
```
Comment (NancyLi1013) [inline ports/pthreadpool/vcpkg.json:3]:
```suggestion
  "version-date": "2020-02-21",
```
It seems more proper to use `version-date` field here. Please see https://github.com/microsoft/vcpkg/blob/master/docs/specifications/versioning.md#22-package-versions

File: versions/p-/pthreadpool.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "04f724a37402046e3d48ec9fe86d4cdfa50df493",
+      "version-string": "2020-02-21",
```
Comment (NancyLi1013) [inline versions/p-/pthreadpool.json:5]:
```suggestion
      "version-date": "2020-02-21",
```
The same as above.

File: ports/pthreadpool/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    OPTIONS
+        -DPTHREADPOOL_BUILD_TESTS=OFF
+        -DPTHREADPOOL_BUILD_BENCHMARKS=OFF
+)
+vcpkg_install_cmake()
+vcpkg_copy_pdbs()
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include
+                    ${CURRENT_PACKAGES_DIR}/debug/bin
```
Comment (NancyLi1013) [inline ports/pthreadpool/portfile.cmake:26]: Why do you remove` ${CURRENT_PACKAGES_DIR}/debug/bin` here?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/pthreadpool/portfile.cmake (hunk 3 above)
Comment (luncliff) [reply ports/pthreadpool/portfile.cmake:26]: This project doesn't have `__declspec(dllexport)` kind of works, so it needs to be `ONLY_STATIC_LIBRARY`.

Comment (luncliff) [reply ports/pthreadpool/portfile.cmake:26]: In short, the line is for `windows` triplets. I think I had to use `if(VCPKG_TARGET_IS_WINDOWS)`.

I will update the part after checking more triplets

Comment (luncliff) [reply ports/pthreadpool/portfile.cmake]: The part is removed in bf52420. :)

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for adding this port @luncliff.

**Pull request merged**